spec/services/customers/manage_invoice_custom_sections_service_spec.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Customers::ManageInvoiceCustomSectionsService do
+  let(:customer) { create(:customer) }
+  let(:invoice_custom_sections) { create_list(:invoice_custom_section, 4, organization: customer.organization) }
+  let(:skip_invoice_custom_sections) { nil }
+  let(:service) { described_class.new(customer: customer, section_ids:, skip_invoice_custom_sections:) }
+  let(:section_ids) { nil }
+
+  before do
+    customer.selected_invoice_custom_sections << invoice_custom_sections[0] if customer
+    customer.organization.selected_invoice_custom_sections = invoice_custom_sections[2..3] if customer
+  end
+
+  describe '#call' do
+    context 'when customer is not found' do
+      let(:customer) { nil }
+
+      it 'returns not found failure' do
+        result = service.call
+        expect(result).not_to be_success
+        expect(result.error).to be_a(BaseService::NotFoundFailure)
+        expect(result.error.message).to eq('customer_not_found')
+      end
+    end
+
+    context 'when sending skip_invoice_custom_sections: true AND selected_ids' do
+      let(:skip_invoice_custom_sections) { true }
+      let(:section_ids) { [1, 2, 3] }
+
+      it 'raises an error' do
+        result = service.call
+        expect(result).not_to be_success
+        expect(result.error).to be_a(BaseService::ValidationFailure)
+        expect(result.error.message).to include('skip_sections_and_selected_ids_sent_together')
+      end
+    end
+
+    context 'when updating selected_invoice_custom_sections' do
+      context 'when section_ids match customer\'s applicable sections' do
```

Comment:
for future reference, our preferred string delimiter is double quotes instead of single quotes, using double quotes won't require to escape the apostrophe character here.